confuse.py

```diff
@@ -889,6 +889,8 @@ def __init__(self, appname, modname=None, read=True):
         super(Configuration, self).__init__([])
         self.appname = appname
         self.modname = modname
+        # Pre-resolve default source location
+        self._package_path = _package_path(appname)
```

Comment:
It looks like this changed the behavior: below `_package_path` if called with `self.modname` not `self.appname`

Reply:
Thanks, I changed this back to using `modname`. The tests failed in that case because `_package_path(None)` doesn't work. I guarded against this with a new check and now things seem to work.